Fix StartingMusic and allow to play music playlist from start

matjaeck opened the pull request:
When you set `StartingMusic` in the defaults the music will on bleed suppress the lobby background music. The change gives the background music priority over the starting music and adds a `PlayFromStart` field to allow starting the playlist with the first track and a `PlayRandom` field to allow starting with a random track. Music will not be played if none of these are enabled and no background or starting music is defined.

matjaeck commented on OpenRA.Mods.Common/Traits/World/MusicPlaylist.cs:
```diff
@@ -79,17 +79,15 @@ public MusicPlaylist(World world, MusicPlaylistInfo info)
-			else
-			{
-				// Start playback with a random song
-				currentSong = random.FirstOrDefault();
```
This disables the fallback to a random song when no background music is set. It's not documented and not used anywhere so I thought this can go away.

GraionDilach replied:
Just because someone disables music shuffling does not mean to penalize him/her with all matches starting with Await and Bigfoot. The randomization is in place to shake that up.

matjaeck replied:
Ok, I just changed this because the description says it would start with the first song, which should then be adjusted. I like the idea that the game always starts with the same song unless the player the changes it, but I will change it if requested.

Commit:
Add myself to authors.

matjaeck commented:
Updated as requested and fixed the description. My intention with this was the opposite though (see #16681), but changing the playback to always start with the first song will probably annoy the people who consider the nostalgia aspect less important.

abcdefg30 commented on OpenRA.Mods.Common/Traits/World/MusicPlaylist.cs:
```diff
@@ -79,17 +79,15 @@ public MusicPlaylist(World world, MusicPlaylistInfo info)
 				CurrentSongIsBackground = false;
 			}
+			else
+			{
```
Please remove the braces around this single-line statement.

matjaeck commented:
Fixed.

matjaeck commented:
I changed the code and added a `PlayFromStart` field which by disabling begins the playlist with a random track, and have set it up this way for TD, RA and D2K. I removed the test case for now.

Commit:
Prioritize BackgroundMusic over StartingMusic.

Commit:
Start music playlist with a random track.

matjaeck commented:
Updated and added a separate `PlayRandom` field to allow disabling the music completely when this and `PlayFromStart` is disabled.

matjaeck commented:
Closing to use my time for testing and might reopen after the release is out and all the high priority refactors / fixes are done.
